### ISSUE TITLE
optimizeFontResourcesDict: missing fontDict entries "BaseFont" and "Name"

### ISSUE THREAD

**Issue by ekanna** [hhrutter/pdfcpu#11]
Hi @hhrutter 

I have printed some PDF files from http://gluon.mxnet.io/ docs
Then i am trying to consolidate those pdfs using below code

```go 
        cmd := pdfcpu.MergeCommand(filenamesIn, outputFile, types.NewDefaultConfiguration())

	err = pdfcpu.Process(&cmd)
	if err != nil {
		logger.Println(err)
		return
	}
```

I am getting this error
```
optimizeFontResourcesDict: missing fontDict entries "BaseFont" and "Name"
```
I am using lastest pdfcpu version v0.1.1 on MacBookPro version 10.12.6

Thanks

**Comment by ekanna** [comment]
I am able to merge those files using qpdf using below command
```
qpdf --empty --pages *.pdf -- out.pdf
```
So i suspect it may be a bug here

**Comment by hhrutter** [comment]
I see.#
Can you paste the stack trace you get with verbose on?

**Comment by ekanna** [comment]
Hi @hhrutter 
Here is the consolidated pdf file.
[gluon-ml-book.pdf](https://github.com/hhrutter/pdfcpu/files/1477447/gluon-ml-book.pdf)
I think now you can split this file and merge it back to find the error(s).

I am not able to split this consolidated file.

Validation command
```
pdfcpu validate pdfs/gluon-ml-book.pdf 
```
Error message
```
validating(mode=relaxed) pdfs/gluon-ml-book.pdf ...
validation error (try -mode=relaxed): validateStreamDictEntry: dict=imageStreamDict entry=SMask unsupported in version 1.3
```
Thank you

**Comment by ekanna** [comment]
Hi @hhrutter 
with pdfcpu version 0.1.2

```
pdfcpu validate pdfs/gluon-ml-book.pdf
 ```
```
validating(mode=relaxed) gluon-ml-book.pdf ...
validation ok
```
But, split command still throws an error

```
pdfcpu split gluon-ml-book.pdf gluonbook/
```
```
splitting gluon-ml-book.pdf into gluonbook/ ...
optimizeFontResourcesDict: missing fontDict entries "BaseFont" and "Name"
```

Verbose
```
pdfcpu split -verbose gluon-ml-book.pdf gluonbook/
```
Extract
```
INFO: 2017/11/18 05:47:10 optimize.go:1000: XRefTable begin
INFO: 2017/11/18 05:47:10 optimize.go:655: optimizeFontAndImages begin
Fatal: optimizeFontResourcesDict: missing fontDict entries "BaseFont" and "Name"
github.com/hhrutter/pdfcpu/optimize.optimizeFontResourcesDict
	/Users/koteswara-0347/goplay/src/github.com/hhrutter/pdfcpu/optimize/optimize.go:207
github.com/hhrutter/pdfcpu/optimize.optimizeResources
	/Users/koteswara-0347/goplay/src/github.com/hhrutter/pdfcpu/optimize/optimize.go:415
github.com/hhrutter/pdfcpu/optimize.parseResourcesDict
	/Users/koteswara-0347/goplay/src/github.com/hhrutter/pdfcpu/optimize/optimize.go:464
github.com/hhrutter/pdfcpu/optimize.parsePagesDict
	/Users/koteswara-0347/goplay/src/github.com/hhrutter/pdfcpu/optimize/optimize.go:531
github.com/hhrutter/pdfcpu/optimize.optimizeFontAndImages
	/Users/koteswara-0347/goplay/src/github.com/hhrutter/pdfcpu/optimize/optimize.go:690
github.com/hhrutter/pdfcpu/optimize.XRefTable
	/Users/koteswara-0347/goplay/src/github.com/hhrutter/pdfcpu/optimize/optimize.go:1003
github.com/hhrutter/pdfcpu.readValidateAndOptimize
	/Users/koteswara-0347/goplay/src/github.com/hhrutter/pdfcpu/api.go:211
github.com/hhrutter/pdfcpu.Split
	/Users/koteswara-0347/goplay/src/github.com/hhrutter/pdfcpu/api.go:462
github.com/hhrutter/pdfcpu.Process
	/Users/koteswara-0347/goplay/src/github.com/hhrutter/pdfcpu/process.go:177
main.main
	/Users/koteswara-0347/goplay/src/github.com/hhrutter/pdfcpu/cmd/pdfcpu/main.go:536
runtime.main
	/usr/local/go/src/runtime/proc.go:185
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:2337
```

**Comment by hhrutter** [comment]
I know - I am on it but right now I'm @GopherConBrasil.

**Comment by hhrutter** [comment]
fixed in version 0.1.2
